### ISSUE TITLE
Wrong characters widths when pages are built separately

### ISSUE THREAD

**Issue by wf-r** [Kozea/WeasyPrint#1489]
Hi,

(this is a separate issue for the comment (https://github.com/Kozea/WeasyPrint/issues/628#issuecomment-960919692) given in #628).

On Version 53.3 I get unexpected horizontal whitespaces using the following code, which I cannot reproduce on 52.5:
```
import jinja2
import weasyprint

pdf_pages = weasyprint.HTML(
    string=jinja2.Environment(
        loader=jinja2.FileSystemLoader(searchpath='.')
    ).get_template(
        'example.html'
    ).render(
        ueberschrift='Lw, UmP, TrD',
    ),
    base_url='.'
).render().pages
pdf_pages.extend(
    weasyprint.HTML(
        string=jinja2.Environment(
            loader=jinja2.FileSystemLoader(searchpath='.')
        ).get_template(
            'example.html'
        ).render(
            ueberschrift='Bw, Offz, GeoInfoD',
        ),
        base_url='.'
    ).render().pages
)

with open('out.pdf', 'w+b') as out:
    weasyprint.Document(
        pages=pdf_pages,
        metadata=weasyprint.document.DocumentMetadata(),
        # font_config=weasyprint.text.fonts.FontConfiguration(),  # for 53.3
        font_config=weasyprint.document.FontConfiguration(),  # for 52.5
        url_fetcher=None,
        # optimize_size='none'  # for 53.3
    ).write_pdf(out)
```

The two respective documents are attached, on 53.3 I get unexpected horizontal whitespaces, on 52.5 it works fine. However, the behaviour is pretty random: 
* if i render only one page in the snippet above, everything is fine
* other strings render fine

[out-52_5.pdf](https://github.com/Kozea/WeasyPrint/files/7496164/out-52_5.pdf)
[out-53_3.pdf](https://github.com/Kozea/WeasyPrint/files/7496165/out-53_3.pdf)

System:

* Weasyprint: 53.3 and 52.5
* Python 3.8.6
* Windows 10 64bit Build 18363
* Pango 1.48.4

If you would like to investigate and need further information please let me know.

Best
Wolfgang

Template used in snippet:
```
<!DOCTYPE html>
<html>
	<head>

		<!-- CSS property to place div
			side by side -->
		<style>
			@page{
				size: A4 landscape;
				margin: 0px;
				padding: 0px;
			}
			html {
				height: 100%;
			}
			body {
				height: 100%;
				display: flex;
				flex-direction: column;
			}
			h1{
				text-align:center; 
			}
			#boxes{
				flex-grow: 1;
				display: flex;
				flex-direction: row;
			}
			.box {
			height: 100%;
				width: 50%;
				margin-left: 20px;
				margin-right: 20px;
				background: #C0C0C0;
				padding-left: 20px;
				padding-right: 20px;
				padding-bottom: 20px;
				border-radius: 30px;
			}
		</style>
	</head>

	<body>
		<h1>{{ ueberschrift }}</h1>
		<div id="boxes">
			<div class="box">
				Some random tetx
            		</div>
			<div class="box">
				Some more random text
        	       </div>
		</div>
	</body>
</html>					 
```

**Comment by trevmono** [comment]
I have experienced exactly the same. I have narrowed it down through trial and error. The last page (or pages in the last render added to the document) always works correctly but every previous pages has unexplained padding or margin between some characters. If I delete the last page from the document before doing write_pdf, the new last page now gets the correct layout and the others before still don't.

If the last item added to the document is multiple pages itself, all of its pages render correctly. For example, if one of the render objects has 10 pages and it is at the end of the combined document, those 10 pages will all work. If I put another single page at the end then all of the pages now don't work.

I am also having problems rendering any italic styles.

So something is wrong with write_pdf and fonts. I noticed there were font changes in the last beta, so the reason will be in there  somewhere. I'm not smart enough to work it out myself, but I've narrowed it down for others.
